server/middleware/auth.go

```diff
@@ -97,14 +97,33 @@ func BypassAuthCaches(ctx context.Context) bool {
 
 func GetJWTValidators(config *config.Config) []*validator.Validator {
 	jwtValidators := make([]*validator.Validator, len(config.Auth.Issuers))
-	for i, issuer := range config.Auth.Issuers {
-		log.Info().Msg(issuer)
-		issuerURL, _ := url.Parse(issuer)
-		provider := jwks.NewCachingProvider(issuerURL, config.Auth.JWKSCacheTimeout)
+	for i, issuerCfg := range config.Auth.Issuers {
+		log.Info().Msg(issuerCfg.Issuer)
+		issuerURL, _ := url.Parse(issuerCfg.Issuer)
+		var keyFunc func(ctx context.Context) (interface{}, error)
+		switch issuerCfg.Algorithm {
+		case validator.RS256:
+			{
+				provider := jwks.NewCachingProvider(issuerURL, config.Auth.JWKSCacheTimeout)
+				keyFunc = provider.KeyFunc
+				break
```

Comment:
nit: You do not need braces and break in go inside cases.

Reply:
made update.